src/include/storage/block_compactor.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include "storage/arrow_block_metadata.h"
+#include "storage/data_table.h"
+#include "storage/storage_defs.h"
+#include "transaction/transaction_manager.h"
+namespace terrier::storage {
+
+/**
+ * Typedef for a standard hash map with varlen entry as the key. The map uses deep equality checks (whether
+ * the stored underlying byte string is the same) for key comparison.
+ */
+template <class T>
+using VarlenEntryMap = std::unordered_map<VarlenEntry, T, VarlenContentHasher, VarlenContentDeepEqual>;
```

Comment:
Move to storage_defs?

Reply:
This alias is internal to the block compactor. We can move when someone else finds a need for it.